ipld/src/cid_hashset.rs

```diff
@@ -8,9 +8,14 @@ use cid::Cid;
 pub struct CidHashSet(HashSet<u64>);
 
 impl CidHashSet {
-    pub fn insert(&mut self, cid: &Cid) -> bool {
+    pub fn insert(&mut self, cid: &Cid, on_inserted: &impl Fn(usize)) -> bool {
```

Comment:
`on_inserted` could be hoisted out of this function, right?

Reply:
Do u mean adding another insert function without the callback parameter?

Reply:
I don't think you need another function. You could just directly call `on_inserted` if `.insert()` returns true.

Reply:
I changed the signature to take advantage of compiler error to make sure I did not miss anything. I can change it back but then the compiler won't complain if we forget to call `on_inserted` when refactoring `recurse_links_hash` fn in the future

Reply:
Okay, that's fine, we can leave it as it is.